examples/edge-functions/supabase/functions/discord-bot/index.ts

```diff
@@ -0,0 +1,150 @@
+// Sift is a small routing library that abstracts away details like starting a
+// listener on a port, and provides a simple function (serve) that has an API
+// to invoke a function for a specific path.
+import { json, serve, validateRequest } from 'https://deno.land/x/sift@0.6.0/mod.ts'
+// TweetNaCl is a cryptography library that we use to verify requests
+// from Discord.
+import nacl from 'https://cdn.skypack.dev/tweetnacl@v1.0.3?dts'
+
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@2'
+
+import { Database } from './types/database.types.ts'
+import { DiscordWebhookEvent } from './types/discord.ts'
+
+const supabaseAdminClient = createClient<Database>(
+  // Supabase API URL - env var exported by default when deployed.
+  Deno.env.get('SUPABASE_URL') ?? '',
+  // Supabase API SERVICE ROLE KEY - env var exported by default when deployed.
+  Deno.env.get('SUPABASE_SERVICE_ROLE_KEY') ?? ''
+)
+
+// For all requests to "/" endpoint, we want to invoke bot() handler.
+serve({
+  '/discord-bot': bot,
+})
+
+// The main logic of the Discord Slash Command is defined in this function.
+async function bot(request: Request) {
+  // validateRequest() ensures that a request is of POST method and
+  // has the following headers.
+  const { error } = await validateRequest(request, {
+    POST: {
+      headers: ['X-Signature-Ed25519', 'X-Signature-Timestamp'],
+    },
+  })
+  if (error) {
+    return json({ error: error.message }, { status: error.status })
+  }
+
+  // verifySignature() verifies if the request is coming from Discord.
+  // When the request's signature is not valid, we return a 401 and this is
+  // important as Discord sends invalid requests to test our verification.
+  const { valid, body } = await verifySignature(request)
+  if (!valid) {
+    return json(
+      { error: 'Invalid request' },
+      {
+        status: 401,
+      }
+    )
+  }
+  console.log(body)
+  const {
+    channel_id = '',
+    type = 0,
+    data = { options: [] },
+    member: { user } = {},
+  } = JSON.parse(body) as DiscordWebhookEvent
+  // Discord performs Ping interactions to test our application.
+  // Type 1 in a request implies a Ping interaction.
+  if (type === 1) {
```

Comment:
minor: would be easier to follow, if we can create enums for different types.

Reply:
Good call, thanks. I've also went ahead and simplified the example.